tests/e2e/raw_block_volume.go

```diff
@@ -596,4 +724,561 @@ var _ = ginkgo.Describe("raw block volume support", func() {
+	   7. Repeat the following steps for all the nodes in the k8s cluster.
+	      a. Drain and Cordon off the node.
+	      b. Enable CSIMigration and CSIMigrationvSphere feature gates on the
+	         kubelet and Restart kubelet.
+	      c. Verify CSI node for the corresponding K8s node has the following
+	         annotation - storage.alpha.kubernetes.io/migrated-plugins.
+	      d. Enable scheduling on the node.
+	   8. Verify all PVC1 and PV1 and have the following annotation -
+	      "pv.kubernetes.io/migrated-to": "csi.vsphere.vmware.com".
+	   9. Verify cnsvspherevolumemigrations crd is created for PVC1 and PV1.
+	   10. Verify CNS entries are present for all PVC1 and PV1 and all PVCs has
+	       correct pod names.
+	   11. Verify data written before migration on PVC1.
+	   12. Create PVC2 using SC1 with volumeMode=Block and wait for binding with PV (say PV2).
+	   13. Verify cnsvspherevolumemigrations crd is created for PVC2 and PV2.
+	   14. Patch DEP1 to use PVC2 as well.
+	   15. Verify CNS entries are present for present for PV2 and PVC2.
+	   16. Verify CNS entries for PVC1 and PVC2 have correct pod names.
+	   17. Write and read some data on raw block PVC PVC2 inside deployment pod.
+	   18. Scale down DEP1 replicas to 0 replicas and wait for PVC1 and PVC2
+	       to detach.
+	   19. Verify CNS entries for PVC1 and PVC2 have pod names removed.
+	   20. Delete DEP1.
+	   21. Wait for PV1 and PV2 and respective vmdks to get deleted.
+	   22. Verify cnsvspherevolumemigrations crds are removed for all PV1, PV2,
+	       PVC1 and PVC2.
+	   23. Verify CNS entries are removed for PV1, PV2, PVC1 and PVC2.
+	   Following steps will be done as a part of AfterEach() call.
+	   24. Delete SC1.
+	   25. Repeat the following steps for all the nodes in the k8s cluster.
+	      a. Drain and Cordon off the node.
+	      b. Disable CSIMigration and CSIMigrationvSphere feature gates on the
+	         kubelet and Restart kubelet.
+	      c. Verify CSI node for the corresponding K8s node does not have the
+	         following annotation - storage.alpha.kubernetes.io/migrated-plugins.
+	      d. Enable scheduling on the node.
+	   26. Disable CSIMigration and CSIMigrationvSphere feature gates on
+	       kube-controller-manager (& restart).
+	*/
+	ginkgo.It("[csi-block-vanilla] [csi-block-vanilla-parallelized] Test VCP-to-CSI migration "+
+		"with raw block volume", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		ginkgo.By("Creating VCP SC")
+		scParameters = make(map[string]string)
+		scParameters[vcpScParamDatastoreName] = GetAndExpectStringEnvVar(envSharedDatastoreName)
+		vcpSc, err := createVcpStorageClass(client, scParameters, nil, "", "", false, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			err := client.StorageV1().StorageClasses().Delete(ctx, vcpSc.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
```

Comment:
we need to test migration with regular and in line volume both

Reply:
yes, currently added tc with regular volume to cover basic functioning, will add tc with inline volume later